extra/resources/SysInfo.in

```diff
@@ -241,12 +241,17 @@ SysInfoStats() {
 SysInfo_megabytes() {
     # Size in megabytes
     echo $1 | awk '{ n = $0;
-                     sub(/[0-9]+(.[0-9]+)?/, "");
-                     split(n, a, $0);
-                     n=a[1];
-                     if ($0 == "G" || $0 == "") { n *= 1024 };
-                     if (/^kB?/) { n /= 1024 };
-                     printf "%d\n", n }' # Intentionally round to an integer
+    sub( /[0-9]+(.[0-9]+)?/, "" );
+    if ( $0 == "" ) { $0 = "B" };
+    split( n, a, $0 );
+    n = a[1];
+    if ( $0 == "P" ) { n *= 1024 * 1024 * 1024 };
+    if ( $0 == "T" ) { n *= 1024 * 1024 };
+    if ( $0 == "G" ) { n *= 1024 };
+    if ( $0 == "M" ) { n *= 1 };
+    if ( $0 == "K" ) { n /= 1024 };
+    if ( $0 == "B" ) { n /= 1024 * 1024 };
```

Comment:
"df -h" on FreeBSD 12 shows e.g. "402G", "1.0K", "0B".

My concern was that POSIX doesn't require -h to be supported at all, and some very old implementations don't have it: https://pubs.opengroup.org/onlinepubs/009695399/utilities/df.html

However, SysInfo has used "df -h" since 2010, and we haven't had any complaints, so we can continue to use it.

Reply:
OK, Thank you!